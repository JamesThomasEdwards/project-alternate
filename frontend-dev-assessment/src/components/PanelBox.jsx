import React from 'react';
// React Bootstrap
import Panel from 'react-bootstrap/lib/Panel';
// Props;
import PropTypes from 'prop-types';

const propTypes = {
    showContent: PropTypes.bool,
    togglePanelContent: PropTypes.func,
    id: PropTypes.number,
    showText: PropTypes.string
};

const PanelBox = (props) => {
    // props.togglePanelContent toggles panel A from;
    // have text (shown by props.showText)/ light background and panel B to have a darkened background;
    // to panel A with darkend background and panel B to have text;
    // it takes in props.id that's given as an index when rendered. This checks that; 
    // id to see if it is true or false in showHidePanelContent;
    return (
        <Panel
            className={props.showContent ? 'panel-show-content' : 'panel-hide-content'}
            onClick={() => props.togglePanelContent(props.id)}>
            <Panel.Body className="panel-container__panel-text">
                {props.showText}
            </Panel.Body>
        </Panel>
    );
};

PanelBox.propTypes = propTypes;

export default PanelBox;