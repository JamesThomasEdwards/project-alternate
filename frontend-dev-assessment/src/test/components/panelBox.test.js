import React from 'react'
import PanelBox from '../../components/PanelBox.jsx';
import ReactShallowRenderer from 'react-test-renderer/shallow';

test('should render PanelBox component correctly', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<PanelBox />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
});