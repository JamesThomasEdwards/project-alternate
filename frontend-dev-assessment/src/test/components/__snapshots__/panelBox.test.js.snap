// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PanelBox component correctly 1`] = `
<Uncontrolled(Panel)
  className="panel-b"
  onClick={[Function]}
>
  <PanelBody
    bsClass="panel"
    className="panel-text"
    collapsible={false}
  />
</Uncontrolled(Panel)>
`;
